langchain-core/src/utils/stream.ts

```diff
@@ -208,7 +220,12 @@ export class AsyncGeneratorWithSetup<
       return this.firstResult;
     }
 
-    return this.generator.next(...args);
+    const storage = AsyncLocalStorageProviderSingleton.getInstance();
+    return new Promise((resolve) => {
```

Comment:
storage.run returns whatever the function returns so no need for the extra promise